pkg/sysinfo/numcpu.go

```diff
@@ -4,12 +4,6 @@ import (
 	"runtime"
 )
 
-// NumCPU returns the number of CPUs. On Linux and Windows, it returns
-// the number of CPUs which are currently online. On other platforms,
-// it's the equivalent of [runtime.NumCPU].
-func NumCPU() int {
-	if ncpu := numCPU(); ncpu > 0 {
-		return ncpu
-	}
-	return runtime.NumCPU()
-}
+// NumCPU returns the number of CPUs. It's the equivalent of [runtime.NumCPU].
+// Deprecated: Use [runtime.NumCPU] instead. It will be removed in the next release.
+var NumCPU = runtime.NumCPU()
```

Comment:
This needs an empty line before the `Deprecated` comment, otherwise it's not detected as "Deprecated".

It's also slightly better to create a wrapper function here; both because we don't want any code from overriding the  exported alias, but also because otherwise it shows up under `variables` on pkg.go.dev, which can be confusing.

```suggestion
// NumCPU returns the number of CPUs. It's the equivalent of [runtime.NumCPU].
//
// Deprecated: Use [runtime.NumCPU] instead. It will be removed in the next release.
func NumCPU() int {
	return runtime.NumCPU()
}
```

Reply:
Ah, good catch, let me update

Reply:
I always forget about the required newline before `Deprecated` 🙈 
Sounds like we should teach our linter to detect that 😅

Reply:
Yes, actually surprised there's no linter picking on those currently!

for the "wrapper vs direct alias"; both options are OK if things get too complicated (sometimes the wrapper means we now have to import other packages, which may not be worth it in some cases), but in this case it was trivial to change.